### ISSUE TITLE
@HostBinding happens before the ngAfterViewInit() lifecycle event has fired

### ISSUE THREAD

**Issue by DavidGangel** [angular/angular#7057]
Hi,

I have this component:

```javascript
import { Component, Input, ContentChildren, QueryList, HostBinding, AfterViewInit } from 'angular2/core';
import { Layer } from './layer.component';

@Component({
    selector: 'citgroupASliderSlide',  
    template: '<ng-content></ng-content>',
    directives: [Layer]
})

export class Slide implements AfterViewInit {   
    
    @Input() private interval:number=5000;
    @Input() private pause:string="hover";
    @Input() private wrap:boolean=true;
    @Input() private keyboard:boolean=true;                  
    
    @Input()
    @HostBinding('class.active')
    public get active() { 
        return this.isActive; 
    }   
    public set active(active:boolean) {
        if(active!=this.isActive) {
            this.isActive=active;
            this.doAnimation();
        }
    } 
            
    @ContentChildren(Layer)        
    private layerComponents: QueryList<Layer>;
    
    private isActive=false;
    private layerComponentsArray: Array<Layer> = [];      

    constructor() {}
    
    ngAfterViewInit() {
        this.layerComponentsArray = this.layerComponents.toArray();                      
    }
    
    private doAnimation():void{
        for(var layer of this.layerComponentsArray) {
            layer.active=this.isActive;
        }
    }                      
}

```

The problem is that the doAnimation() method gets called through the set active(active:boolean) method before the ngAfterViewInit() method.

**Comment by ericmartinezr** [comment]
related https://github.com/angular/angular/issues/4748

**Comment by vicb** [comment]
This will be solved via #6270
